src/typemap.c

```diff
@@ -1056,7 +1068,7 @@ jl_typemap_entry_t *jl_typemap_insert(union jl_typemap_t *cache, jl_value_t *par
             newrec->isleafsig = newrec->issimplesig = 0;
     }
```

Comment:
Might as well add an executable assertion on the meaning of `simplekeys` here:
`assert((!simplekeys || newrec->issimplesig) && "bad insert")`